### PR TITLE
Add reclaim space protobuf file

### PR DESCRIPTION
**Pull request by yati1998**
This commit adds the protobuf file which includes the spec
for communication between the CSIAddons Controller and sidecar.

Signed-off-by: yati1998 <ypadia@redhat.com>

### CODE REVIEW

**Review by Rakshith-R — CHANGES_REQUESTED** [review]
Let this package go inside reclaimspace package like
`internal/service/reclaimspace` to better accommodate other services.

**Comment by Rakshith-R** [comment]
Please add a new makefile to generate go files from protoc files or modify existing makefile or both.
Thanks

**Comment by yati1998** [comment]
> Please add a new makefile to generate go files from protoc files or modify existing makefile or both. Thanks

After talking to @nixpanic , he suggested to directly write the protobuf file. @nixpanic What's your thought here? shall I write the README to create the protobuf?

File: internal/service/reclaimspace/reclaimspace.proto [hunk 1]
```diff
@@ -0,0 +1,49 @@
+syntax = "proto3";
+package service;
+
+option go_package = "github.com/csi-addons/kubernetes-csi-addons/internal/service";
```
Comment (Rakshith-R) [inline internal/service/reclaimspace/reclaimspace.proto:4]: This needs to be updated.

**Review by Rakshith-R — CHANGES_REQUESTED** [review]
Just makefile with commands to generate go files.

**Comment by nixpanic** [comment]
> > Please add a new makefile to generate go files from protoc files or modify existing makefile or both. Thanks
> 
> After talking to @nixpanic , he suggested to directly write the protobuf file. @nixpanic What's your thought here? shall I write the README to create the protobuf?

There is no need to place the `protobuf` parts in a readme. This is internal to this repository only, not shared for some external API.

There might be a benefit in automatically re-generating the go files from the `.proto` file, but I do not think we need to do that now. If we really think it helps, we can always add such automation later.

**Comment by nixpanic** [comment]
> Just makefile with commands to generate go files.

The problem is installing the `protoc` components... The way it is done in csi-addons/spec is not clean, so I prefer manual generation before we add automation around this and other parts of the project.

File: internal/service/reclaimspace/reclaimspace.proto (hunk 1 above)
Comment (nixpanic) [reply internal/service/reclaimspace/reclaimspace.proto:4]: updated to what?

File: internal/service/reclaimspace/reclaimspace.proto [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+// communication between the CSIAddons controller and the sidecar.
+service ReclaimSpace {
+    // ControllerReclaimSpace is a procedure that gets called on the CSI
+    // sidecar.
+    rpc ControllerReclaimSpace (ControllerReclaimSpaceRequest) returns(ReclaimSpaceResponse) {}
+    // NodeReclaimSpace is a procedure that gets called on the CSI
+    // sidecar.
+    rpc NodeReclaimSpace (NodeReclaimSpaceRequest) returns(ReclaimSpaceResponse) {}
+
+  }
```
Comment (nixpanic) [inline internal/service/reclaimspace/reclaimspace.proto:16]: formatting is a little weird, please check indention and add empty lines after a closing `}`

File: internal/service/reclaimspace/reclaimspace.proto [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+// ControllerReclaimSpaceRequest contains the information i.e., pv_name
+// received from the CSIAddons controller for performing the reclaim operation.
+message ControllerReclaimSpaceRequest {
+    // The name of the pv. This field is REQUIRED.
+    string pv_name = 1;
+}
+
+// SidecarCSIAddonsResponse holda the information about the result of the
```
Comment (nixpanic) [inline internal/service/reclaimspace/reclaimspace.proto:24]: Please review the comments, `SidecarCSIAddonsResponse` is not correct (and `holda` typo)

File: internal/service/reclaimspace/reclaimspace.proto [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    // The name of the pv. This field is REQUIRED.
+    string pv_name = 1;
+}
+
+// StorageConsumption contains the usage in bytes.
+message StorageConsumption {
+    // This field is REQUIRED. usage_bytes contains the consumed storage in
+    // bytes.
+    int64 usage_bytes = 1;
+  }
```
Comment (nixpanic) [inline internal/service/reclaimspace/reclaimspace.proto:48]: indention?

**Review by nixpanic — CHANGES_REQUESTED** [review]

File: internal/service/reclaimspace/reclaimspace.proto [hunk 5]
```diff
@@ -0,0 +1,49 @@
+syntax = "proto3";
+package service;
```
Comment (Rakshith-R) [inline internal/service/reclaimspace/reclaimspace.proto:2]: change this too and regenerate the proto files.
```suggestion
package reclaimspace;
```

**Review by Rakshith-R — CHANGES_REQUESTED** [review]

**Review by Rakshith-R — COMMENTED** [review]
Looks good to me, I'll wait for @nixpanic review

**Review by nixpanic — APPROVED** [review]

**Review by Rakshith-R — APPROVED** [review]

**Pull request merged**